tests/test_ignore.py

```diff
@@ -0,0 +1,68 @@
+import os
+
+from dvc.ignore import DvcIgnoreFromFile, DvcIgnore, DvcIgnoreFileHandler
+from dvc.utils.compat import cast_bytes
+from tests.basic_env import TestDvc
+
+
+class TestDvcIgnore(TestDvc):
+    def setUp(self):
+        super(TestDvcIgnore, self).setUp()
+        self.ignore_file_handler = DvcIgnoreFileHandler(self.dvc.tree)
+
+    def _get_all_paths(self):
+
+        paths = []
+        ignore_file_handler = DvcIgnoreFileHandler(self.dvc.tree)
+        for root, dirs, files in self.dvc.tree.walk(
+            self.dvc.root_dir, ignore_file_handler=ignore_file_handler
+        ):
+            for dir in dirs:
```

Comment:
Please don't use `dir` and `file` as var names, because they highlight as builtins.